### PR TITLE
[2473] General readability cleanup (part 1)

### PR DESCRIPTION
**Pull request by faissalMT**
### Context
In order to be able to deliver value it must be easy to understand and work with the code.

### Changes proposed in this pull request
This refactors any unclear or non-idiomatic lines.

### Guidance to review

This PR focuses on Course, its concerns and the jobs.  

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by faissalMT** [comment]
[Hunt down refactoring spring clean](https://trello.com/c/hmQmLF5n/2454-hunt-down-refactoring-spring-clean)

**Comment by faissalMT** [comment]
[Hunt down refactoring spring clean](https://trello.com/c/hmQmLF5n/2454-hunt-down-refactoring-spring-clean)

**Comment by faissalMT** [comment]
[General readability improvements](https://trello.com/c/RUWXGqPw/2473-general-readability-improvements)

File: app/models/course.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Course < ApplicationRecord
   after_initialize :set_defaults
 
   before_discard do
-    raise "You cannot delete a running course (Course: #{self}, Provider: #{provider_id})" if ucas_status != :new
+    raise "You cannot delete a running course (Course: #{self}, Provider: #{provider_id})" unless ucas_status == :new
```
Comment (defong) [inline app/models/course.rb]: dont `self` uses the `to_s ` that gives provider stuff as well?

Comment (faissalMT) [reply app/models/course.rb]: Yeah self will use the to_s

Comment (defong) [reply app/models/course.rb]: can you make the message nice please...

**Pull request merged**